include/seqan3/alphabet/aminoacid/translation.hpp

```diff
@@ -0,0 +1,187 @@
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \ingroup aminoacid
+ * \author Sara Hetzel <sara.hetzel AT fu-berlin.de>
+ * \brief Contains functions for translating a triplet of nucleotides into an amino acid
+ */
+
+#pragma once
+
+#include <tuple>
+
+#include <seqan3/alphabet/aminoacid/aa27.hpp>
+#include <seqan3/range/concept.hpp>
+
+namespace seqan3
+{
+/*!\brief Genetic codes used for translation of nucleotides into amino acids
+ *
+ * \details
+ * The numeric values of the enums correspond to the genbank transl_table values
+ * (see "http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi").
+ */
+enum struct genetic_code : uint8_t
+{
+    CANONICAL=1,
+//     VERT_MITOCHONDRIAL,
+//     YEAST_MITOCHONDRIAL,
+//     MOLD_MITOCHONDRIAL,
+//     INVERT_MITOCHONDRIAL,
+//     CILIATE,
+//     FLATWORM_MITOCHONDRIAL = 9,
+//     EUPLOTID,
+//     PROKARYOTE,
+//     ALT_YEAST,
+//     ASCIDIAN_MITOCHONDRIAL,
+//     ALT_FLATWORM_MITOCHONDRIAL,
+//     BLEPHARISMA,
+//     CHLOROPHYCEAN_MITOCHONDRIAL,
+//     TREMATODE_MITOCHONDRIAL = 21,
+//     SCENEDESMUS_MITOCHONDRIAL,
+//     THRAUSTOCHYTRIUM_MITOCHONDRIAL,
+//     PTEROBRANCHIA_MITOCHONDRIAL,
+//     GRACILIBACTERIA
+};
+}
+
+#include <seqan3/alphabet/aminoacid/translation_details.hpp>
+
+namespace seqan3
+{
+/*!\brief Translate one nucleotide triplet into single amino acid (single nucleotide interface)
+ *
+ * \param n1 First nucleotide in triplet
+ * \param n2 Second nucleotide in triplet
+ * \param n3 Third nucleotide in triplet
+ *
+ * \details
+ *
+ * Translates single nucleotides into amino acid according to given genetic code.
+ *
+ * \par Complexity
+ *
+ * Constant.
+ *
+ * \par Exceptions
+ *
+ * No-throw guarantee.
+*/
+template <nucleotide_concept nucl_type, genetic_code gc = genetic_code::CANONICAL>
+constexpr aa27 translate_triplet(nucl_type const & n1, nucl_type const & n2, nucl_type const & n3) noexcept
+{
+    return seqan3::detail::translation_table<nucl_type, gc>::VALUE[n1.to_rank()][n2.to_rank()][n3.to_rank()];
```

Comment:
Please use the free function interface, i.e. `to_rank(n1)`, instead if the member function (we implement member functions for most things, but this function requires the nucleotide_concept which only requires free functions so you can only assume these exist.